src/raftstore/store/store.rs

```diff
@@ -1117,7 +1117,18 @@ impl<T: Transport, C: PdClient> Store<T, C> {
                     panic!("{} failed to save append state result: {:?}", self.tag, e);
                 });
         }
+
         fail_point!("raft_between_save");
+        fail_point!(
+            "raft_snapshot_between_save",
+            append_res
+                .iter()
+                .any(|&(_, ref ctx)| ctx.snap_region.is_some()),
+            |_| {
+                self.sendch.send(Msg::Quit).unwrap();
```

Comment:
If we only return error here, the event loop is still running so that the node can receive and process other commands or messages.

Reply:
here I guess you just want to snapshot save failed, so I am confused why we need to quit the raftstore here.

Reply:
Because we have 2 logic to test in this case: 1 for **snapshot save failed**, and one for **recover snapshot when TiKV start**. So I need to stop and restart the TiKV.

Reply:
here you just stop the raftstore thread, not TiKV. 

If you want to stop and restart the TiKV, I suggest doing outside explicitly, not here. Here is just to ingest failure to snapshot.